### PR TITLE
CM-455: Add new validation for slug fields

### PR DESCRIPTION
**Pull request by ayumi-oxd**
### Jira Ticket:
CM-455

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/CM-455

### Description:
- Added new validation for slug fields to prevent having a slug such as `-` or `-/-`

### CODE REVIEW

File: src/cms/config/functions/slugValidator.js [hunk 1]
```diff
@@ -22,6 +22,12 @@ module.exports = {
       throw strapi.errors.badRequest('Do not add slash to the beginning of slugs.');
     }
   },
+  // a leading dash is not allowed
+  slugNoLeadingDashValidator: function (slug) {
+    if (slug?.slice(0, 1) === '-') {
+      throw strapi.errors.badRequest('Do not add dash to the beginning of slugs.');
+    }
+  },
```
Comment (molund) [inline src/cms/config/functions/slugValidator.js:30]: you could include a second check
```
if (slug.includes('/-')) {
      throw strapi.errors.badRequest('Do not add dash after a slash.');
}
```

File: src/cms/config/functions/slugValidator.js [hunk 2]
```diff
@@ -34,6 +40,12 @@ module.exports = {
       throw strapi.errors.badRequest('Do not add slash to the end of slugs.');
     }
   },
+  // a trailing dash is not allowed
+  slugNoTrailingDashValidator: function (slug) {
+    if (slug?.slice(-1) === '-') {
+      throw strapi.errors.badRequest('Do not add dash to the end of slugs.');
+    }
```
Comment (molund) [inline src/cms/config/functions/slugValidator.js:47]: you could include a second check

if (slug.includes('-/')) {
      throw strapi.errors.badRequest('Do not add dash before a slash.');
}

File: src/cms/config/functions/slugValidator.js (hunk 1 above)
Comment (molund) [reply src/cms/config/functions/slugValidator.js:33]: You'll need to add slugNoLeadingDashValidator back onto "page" for this work now for 
`/-about/foo`

File: src/cms/api/page/models/page.js [hunk 3]
```diff
@@ -12,10 +12,14 @@ module.exports = {
     beforeCreate: async (data) => {
       validator.slugCharacterValidator(data.Slug)
       validator.slugLeadingSlashValidator(data.Slug)
+      validator.slugNoLeadingDashValidator(data.Slug)
```
[the remark this answers is not in the document]
Comment (molund) [reply src/cms/api/page/models/page.js]: sorry, this comment caused the test to fail for the other comment

[the remark this answers is not in the document]
Comment (ayumi-oxd) [reply src/cms/api/page/models/page.js:15]: No worries, thank you for testing!

**Review by molund — APPROVED** [review]

**Review by nataliasabadysh — APPROVED** [review]

**Pull request merged**